primitives/src/hash_types/generic.rs

```diff
@@ -28,10 +28,49 @@ super::impl_serde!(HashType, LEN);
 super::impl_bytelike_traits!(HashType, LEN);
 
 #[cfg(feature = "hex")]
-hex::impl_fmt_traits! {
-    #[display_backward(REVERSE)]
-    impl fmt_traits for HashType {
-        const LENGTH: usize = LEN;
+macro_rules! impl_hashtype_fmt_traits {
```

Comment:
We want to put these in `mod.rs` mate. That way we only have a single copy, where as, as it is here, the macro definition is being duplicated by the call to `include!`